kuma/static/js/wiki-edit.js

```diff
@@ -458,90 +458,120 @@
+        // save and edit attempts ajax submit
+        $('.btn-save-and-edit').on('click', function(event) {
+            // disable form
+            $('#wiki-page-edit').attr('disabled', true);
 
-            notifications.push(mdn.Notifier.growl('Saving changes…', { duration: 0 }));
+            // give user feedback
+            var saveNotification = mdn.Notifier.growl('Saving changes…', { duration: 0 });
 
+            // record event
             mdn.analytics.trackEvent({
-                category: 'Wiki',
-                action: 'Button',
-                label: 'Save and Keep Editing' // now "Publish and keep editing" but keeping label for analytics continuity
-            });
+                    category: 'Wiki',
+                    action: 'Button',
+                    label: 'Save and Keep Editing'
+                });
 
+            // Preserve editor content incase something goes wrong
             var savedTa = $form.find('textarea[name=content]').val();
-            if (draftingEnabled) {
-                // Preserve editor content, because saving to the iframe can
-                // yield things like 403 / login-required errors that bust out
-                // of the frame
-                manualSaveDraft(savedTa);
-                startingContent = savedTa;
+            if (supportsLocalStorage) {
+                saveDraft(savedTa);
                 clearTimeout(DRAFT_TIMEOUT_ID);
             }
-            // Redirect the editor form to the iframe.
-            $form.attr('action', '?iframe=1').attr('target', 'save-and-edit-target');
-            return true;
-        });
-        $('.btn-save-and-edit').show();
-
-        $('#save-and-edit-target').on('load', function () {
-            if(notifications[0]) notifications[0].success(null, 2000);
-            notifications.shift();
-
-            var if_doc = $(this).get(0).contentDocument;
-            if (typeof(if_doc) != 'undefined') {
 
-                var ir = $('#iframe-response', if_doc);
-                if ('OK' == ir.attr('data-status')) {
+            // get form data
+            var formData = $form.serialize();
+            var formURL = window.location.href; // submits to self
+
+            $.ajax({
+                url : formURL + '?async',
+                type: "POST",
+                data : formData,
+                dataType : 'html',
+                success: function(data, textStatus, jqXHR) {
+                    // server came back 200
+                    // was there an error, or did the session expire?
+                    var $parsedData = $($.parseHTML(data));
+                    var $responseErrors = $parsedData.find('.errorlist');
+                    var $responseLoginRequired = $parsedData.find('#login');
+                    // If there are errors, saveNotification() for each of them
+                    if ($responseErrors.length) {
+                        var liErrors = $responseErrors.find('li')
```

Comment:
We like to use a `$` in front of variables that contain jQuery objects. Please change liErrors to $liErrors.